src/App.tsx

```diff
@@ -17,18 +53,31 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                rule={rule}
+                setQuery={setQuery}
+                chooseRule={setRule}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {!isLoading && (
+                <TodoList todos={filteredTodos} handleClick={handleClick} />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalLoading && (
```

Comment:
```suggestion
      {selectedTodo && (
```
You can show the modal when you have selected todo